File: app/src/main/java/zlc/season/rxdownloadproject/BaseApplication.java
package zlc.season.rxdownloadproject;

import android.app.Application;

import com.github.anrwatchdog.ANRWatchDog;

/**
 * Author: Season(ssseasonnn@gmail.com)
 * Date: 2016/10/27
 * Time: 13:29
 * FIXME
 */

public class BaseApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        new ANRWatchDog().start();
    }


}

File: demo/src/main/java/zlc/season/rxdownload/demo/BaseApplication.kt
package zlc.season.rxdownload.demo

import android.app.Application
import android.content.Context
import android.support.multidex.MultiDex
import io.reactivex.exceptions.UndeliverableException
import io.reactivex.plugins.RxJavaPlugins
import zlc.season.rxdownload4.utils.log


class BaseApplication : Application() {

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

    override fun onCreate() {
        super.onCreate()

        RxJavaPlugins.setErrorHandler {
            if (it is UndeliverableException) {
                //do nothing
            } else {
                it.log()
            }
        }
    }
}